fgsmk/log.py

```diff
@@ -62,6 +71,21 @@ def many_from_log_file(cls, log_dir: Path, snakemake_log: Path) -> list["RuleLog
             assert len(lines) > 1, (
                 f"Expected at least one line in the Snakemake log file after:\n{lines[0]}"
             )
+
+            # Extract jobid for deduplication
+            jobid: str | None = None
+            for line in lines[1:5]:  # Check next few lines for jobid
+                if line.startswith(cls.JOBID_PREFIX):
+                    jobid = line[len(cls.JOBID_PREFIX) :].strip()
```

Comment:
Why not use [`str.lstrip()`](https://docs.python.org/3/library/stdtypes.html#str.lstrip)?